frontend/app/api/farcaster/[userId]/route.ts

```diff
@@ -0,0 +1,23 @@
+import { publishNewUserKeysCast } from "@/lib/api/backend/farcaster";
+import { ERRORS } from "@/lib/errors";
+import prisma from "@/lib/prisma";
+import { SocialProfileType } from "@prisma/client";
+
+export async function POST(req: Request, { params }: { params: { userId: number } }) {
+  if (process.env.NODE_ENV !== "production") {
+    const userFarcaster = await prisma.socialProfile.findUniqueOrThrow({
+      where: {
+        userId_type: {
+          userId: params.userId,
+          type: SocialProfileType.FARCASTER
+        }
+      }
+    });
+    if (!userFarcaster) {
+      return Response.json({ error: ERRORS.USER_NOT_ON_FARCASTER }, { status: 400 });
+    }
+    const user = await prisma.user.findUniqueOrThrow({ where: { id: params.userId } });
+    await publishNewUserKeysCast(userFarcaster.profileName, `https://app.builder.fi/profile/${user.wallet}`);
+  }
+  return Response.json({ success: true });
```

Comment:
is there any information we can return to the frontend to create a better experience? what do we get from farcaster? I think being able to show a link to the user to see their post could be cool?

Reply:
Yep can do!